test/python/qpy/test_circuit_load_from_qpy.py

```diff
@@ -289,27 +290,39 @@ def test_rdiv(self):
         self.assert_roundtrip_equal(qc)
 
 
+@ddt
 class TestUseSymengineFlag(QpyCircuitTestCase):
     """Test that the symengine flag works correctly."""
 
-    def test_use_symengine_with_bool_like(self):
+    @data(True, False)
+    def test_use_symengine_with_bool_like(self, use_symengine):
         """Test that the use_symengine flag is set correctly with a bool-like input."""
+        if use_symengine and not optionals.HAS_SYMENGINE:
+            raise unittest.skipTest("we can't force 'use_symengine' if we don't have it")
```

Comment:
FWIW, this flag doesn't actually do anything with QPY >13 (which are the only supported ones in Qiskit 2.x) it's just ignored metadata in the payload header. So I'm not sure why we would need symengine installed for this to work.